### PR TITLE
Add profiler agent

### PR DESCRIPTION
**Pull request by george-edgedelta**
@taylanisikdemir Could you please review this document about adding a profiler during agent installation. I quickly threw this together, so I wanted to make sure that it is technically accurate.

### CODE REVIEW

**Comment by george-edgedelta** [comment]
@taylanisikdemir Could you please review this document about adding a profiler during agent installation. I quickly threw this together, so I wanted to make sure that it is technically accurate.

File: docs/installation/add-profiler-agent.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+Overview
+--------
+
+You can use this document to learn how to deploy the Edge Delta Agent with a profiler.
+
+> **Note**
+>
+> This document only applies to Go-based web applications.
```
Comment (taylanisikdemir) [inline docs/installation/add-profiler-agent.md:8]: This is confusing and not fully accurate. let's remove this note

File: docs/installation/add-profiler-agent.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+For Go-based web applications, you can add a profiler during agent deployment. A profiler is useful to obtain data about instances where the agent is deployed, such as reviewing why CPU usage is high.
```
Comment (taylanisikdemir) [inline docs/installation/add-profiler-agent.md:10]: This is also not fully accurate as a generic statement about go based web applications. let's change first sentence to
"Edge Delta agent is a go based binary and it exposes cpu/memory profiling information if specified during deployment. "

File: docs/installation/add-profiler-agent.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+**Linux Example**
+
+For instance, for **Linux**, the regular deployment command is: 
+
+<code>ED_API_KEY=f1111e-e1d1-1ad1-b11d-d1a11111b1 bash -c "$(curl -L https://release.edgedelta.com/release/install.sh)"</code>
+
+To deploy with a profiler, the command would be:
+
+<code>PROFILER_PORT=6060 ED_API_KEY=f1111e-e1d1-1ad1-b11d-d1a11111b1 bash -c "$(curl -L https://release.edgedelta.com/release/install.sh)"</code>
```
Comment (taylanisikdemir) [inline docs/installation/add-profiler-agent.md:35]: this should be change with below:
```
ED_ENV_VARS=PROFILER_PORT=6060 ED_API_KEY=f1111e-e1d1-1ad1-b11d-d1a11111b1 bash ...
```

Comment (taylanisikdemir) [reply docs/installation/add-profiler-agent.md:35]: @albert-edgedelta please confirm

File: docs/installation/add-profiler-agent.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Obtain a Profiler Report 
+-------------------------
+
+When you obtain a profiler report, you can send the report to Edge Delta for analysis. 
+
+Based on the information you want to obtain, run the following command: 
+
+For information on Heap, run: 
+
+<code>curl -sK -v <http://localhost:8080/debug/pprof/heap> > heap.out</code>
```
Comment (taylanisikdemir) [inline docs/installation/add-profiler-agent.md:68]: change the extensions with .pb.gz

File: docs/installation/add-profiler-agent.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+For information on CPU, run:
+
+<code>curl <http://localhost:6060/debug/pprof/profile?seconds=60> --output /tmp/cpu.pb.gz</code>
+
+For information on goroutine, run: 
+
+  <code>curl <http://localhost:6060/debug/pprof/goroutine> > goroutine.out</code>
+
```
Comment (taylanisikdemir) [inline docs/installation/add-profiler-agent.md:77]: the profile dump data can be inspected/visualized using [pprof tool](https://github.com/google/pprof).

install pprof (requires golang installed): `go install github.com/google/pprof@latest`

visualize pprof result: `pprof -web /tmp/cpu.pb.gz`

File: docs/installation/add-profiler-agent.md (hunk 3 above)
Comment (albert-edgedelta) [reply docs/installation/add-profiler-agent.md:29]: You're right @taylanisikdemir, the profiler port needs to be called with ED_ENV_VARS for the agent to inherit the env vars correctly.

**Pull request merged**